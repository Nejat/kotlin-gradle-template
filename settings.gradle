include 'spring-approvals-testing'
include 'spring-boot-jar'
include 'spring-web-client'
include 'spring-rest-api'
include 'common-library'

