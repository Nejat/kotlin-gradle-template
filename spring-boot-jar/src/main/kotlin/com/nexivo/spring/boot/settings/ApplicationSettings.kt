package com.nexivo.spring.boot.settings

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.stereotype.Component

@Component
@ConfigurationProperties("app-settings")
@EnableConfigurationProperties(ApplicationSettings::class)
class ApplicationSettings {

    var number: Double = 0.0

    lateinit var string: String

    val map: MutableMap<String, SomePojo> = mutableMapOf()

    val list: MutableList<String> = mutableListOf()

    class SomePojo {

        lateinit var player1: String

        lateinit var player2: String

        lateinit var toy: String
    }
}