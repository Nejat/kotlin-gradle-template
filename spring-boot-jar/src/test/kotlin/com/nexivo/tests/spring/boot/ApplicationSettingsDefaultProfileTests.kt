package com.nexivo.tests.spring.boot

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.SerializationFeature
import com.fasterxml.jackson.module.kotlin.registerKotlinModule
import com.nexivo.spring.boot.settings.ApplicationSettings
import org.approvaltests.Approvals
import org.approvaltests.reporters.DiffReporter
import org.approvaltests.reporters.UseReporter
import org.junit.Test
import org.junit.runner.RunWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.junit4.SpringRunner

@RunWith(SpringRunner::class)
@SpringBootTest(classes = [ApplicationSettings::class])
@UseReporter(DiffReporter::class)
class ApplicationSettingsDefaultProfileTests {

    @Autowired
    private lateinit var applicationSettings: ApplicationSettings

    private val mapper: ObjectMapper = ObjectMapper().registerKotlinModule()


    init {
        mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL)
        mapper.enable(SerializationFeature.INDENT_OUTPUT)
    }

    @Test
    fun test_application_settings_default_profile () {

        val json = mapper.writeValueAsString(applicationSettings)

        Approvals.verify(json)
    }
}

