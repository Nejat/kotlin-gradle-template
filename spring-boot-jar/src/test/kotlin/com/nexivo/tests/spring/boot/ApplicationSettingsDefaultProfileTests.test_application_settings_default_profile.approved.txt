{
  "number" : 41.99998,
  "string" : "foo",
  "map" : {
    "hill" : {
      "player1" : "Jack",
      "player2" : "Hill",
      "toy" : "Pail"
    },
    "phapha" : {
      "player1" : "Fooey",
      "player2" : "Looey",
      "toy" : "Pen"
    }
  },
  "list" : [ "value1", "value2", "value3" ]
}